dev/NumberBox/NumberBoxParser.h

```diff
@@ -0,0 +1,65 @@
+﻿#pragma once
+#include <vector>
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <cmath>
+#include <algorithm>
+#include <iterator>
+#include <cstdlib>
+#include <regex>
+#include <stack>
+
+// Represents a token in a mathematic expression. This can be a simple operator (+-*/^), a decimal number, or parentheses. 
+class MathToken
+{
+public:
+    enum TokenType
+    {
+        Invalid = 0,
+        Numeric = 1,
+        Operator = 2,
+        ParenLeft = 3,
+        ParenRight = 4,
+        EOFToken = 6
+    };
+    TokenType type;
+    std::string str;
+    MathToken();
+    MathToken(TokenType t, std::string s);
+};
+
+// Handles tokenizing strings
+class MathTokenizer
+{
+    private:
+        std::string inputString;
+        int inputLength;
+        int index;
+        bool IsNumeric(std::string in);
+        bool IsOperator(std::string ss);
+
+    public:
+        MathTokenizer(std::string input);
```

Comment:
recommend passing by && so that temporaries can be `std::move`-d in to the member.